tests/test_config.py

```diff
@@ -172,6 +180,36 @@ def test_parse_config(tmp_path: Path) -> None:
     # ImageVersion= is not set explicitly anymore, so now the version from mkosi.version should be used.
     assert config.image_version == "1.2.3"
 
+    (tmp_path / "abc").mkdir()
+    (tmp_path / "abc/mkosi.conf").write_text(
+        """\
+        [Content]
+        Bootable=yes
+        """
+    )
+    (tmp_path / "abc/mkosi.conf.d").mkdir()
+    (tmp_path / "abc/mkosi.conf.d/abc.conf").write_text(
+        """\
+        [Output]
+        SplitArtifacts=yes
+        """
+    )
+
+    with chdir(tmp_path):
+        _, [config] = parse_config()
+        assert config.bootable == ConfigFeature.auto
+        assert config.split_artifacts == False
+
+        # Passing the directory should include both the main config file and the dropin.
+        _, [config] = parse_config(["--include", os.fspath(tmp_path / "abc")])
+        assert config.bootable == ConfigFeature.enabled
+        assert config.split_artifacts == True
+
+        # Passing the main config file should not include the dropin.
+        _, [config] = parse_config(["--include", os.fspath(tmp_path / "abc/mkosi.conf")])
```

Comment:
I'm not quite sure about the ergonomics of this, i.e. it seems easy to reason wrongly about this, since I always expect dropins to be parsed. How about this: When a directory is passed, we'll look for `mkosi.conf` by defeault. If a file is passed, we take it, but in either case all dropins are always honoured, i.e.

- `--include abc` -> looks for `abc/mkosi.conf` and all `abc/mkosi.conf.d/*.conf`
- `--include abc/qux.foo` looks for `abc/qux.foo` and all `abc/qux.conf.d/*conf`

Reply:
So this would come down to either parsing the directory of any given files or refusing files and always requiring users to pass directories.

Currently we only parse from mkosi.conf.d/ and I think it makes sense to keep that as it is instead of starting to parse arbitrary dropins to keep things simple. If users want dropins, they have to put their file in a directory and include that directory. I don't think that's a very limiting restriction.

Reply:
It is not very limiting, but I was thinking about our spanking new config file spec and the lin
> All configuration paths must support drop-ins.

which we should aspire to.

Reply:
Well, we do support drop-ins, by making the file a directory, putting the file as mkosi.conf in the directory and putting the dropins in mkosi.conf.d/.

Reply:
We can probably still do this in some form, but it'd be in addition to what we already have, so I'd like to leave that for another PR